src/pages/api/v3/push.ts

```diff
@@ -153,6 +153,7 @@ export default async function handler(
                 },
                 data: {
                   lastMutationId,
+                  version: nextVersion,
```

Comment:
This was the bug. Client records' version wasn't getting bumped, so pull wasn't returning their lmid.